added second outer circle and added more control how to color them

Nopileos2 opened the pull request:
No idea if this goes beyond the idea of this plugin. I initially wrote this for me with the main intention to get a better feeling and visual representation what my melee range is. So I added a second outer circle (changeable in size) and also added the option to decouple the color of the circles from the guidelines.

I tested config compatibility and for me it seemed to work going from the current config version to a new one, also tried simulating with even older config without the target circle.

I had to extract the circle drawing into a function which has quite a lot arguments now. Not really happy about it but did not see a good alternative without reworking more parts, which I wanted to avoid.

Nopileos2 committed:
added second outer circle and added more control how to color them

VariableVixen commented on PositionalGuide/ConfigWindow.cs:
```diff
@@ -181,6 +197,24 @@ public class ConfigWindow: Window, IDisposable {
+
+		ImGui.Indent();
+		changed |= ImGui.Checkbox("Always use the defined color for the target circle?", ref circleColours[1]);
+		utils.Tooltip("As above but only for the target circle.");
```
Please swap the tooltip call and the unindent call to match the rest of the rendering.

VariableVixen commented on PositionalGuide/ConfigWindow.cs:
```diff
@@ -181,6 +197,24 @@ public class ConfigWindow: Window, IDisposable {
+
+		ImGui.Indent();
+		changed |= ImGui.Checkbox("Always use the defined color for the outer circle?", ref circleColours[2]);
+		utils.Tooltip("As above but only for the outer circle.");
```
Swap these two calls as well

VariableVixen commented on PositionalGuide/ConfigWindow.cs:
```diff
@@ -232,7 +269,13 @@ public class ConfigWindow: Window, IDisposable {
+			ImGui.Unindent();
+			ImGui.TextUnformatted("Targets for circles are:");
+			ImGui.Indent();
+			ImGui.TextUnformatted("For the target circle: c, circle, circle_target");
```
All other names use `-` instead of `_`, please maintain that for more intuitive use. Maybe `target-circle` to fit the english-y format of the others, using common speech ordering?

VariableVixen commented on PositionalGuide/ConfigWindow.cs:
```diff
@@ -232,7 +269,13 @@ public class ConfigWindow: Window, IDisposable {
+			ImGui.TextUnformatted("Targets for circles are:");
+			ImGui.Indent();
+			ImGui.TextUnformatted("For the target circle: c, circle, circle_target");
+			ImGui.TextUnformatted("For the outer circle: co, circle_outer, outer");
```
Likewise here, change `_` to `-` and maybe swap it to be `outer-circle` instead.

VariableVixen commented on PositionalGuide/Plugin.cs:
```diff
@@ -373,7 +401,17 @@ private static IEnumerable<Vector3> arcPoints(Vector3 centre, Vector3 start, dou
 				break;
 			case "c":
 			case "circle":
+			case "circle_target":
```
Remember to change the `case` values for the circle toggles

VariableVixen commented on PositionalGuide/PositionalGuide.csproj:
```diff
@@ -2,7 +2,7 @@
 	<PropertyGroup>
 		<Product>PositionalGuide</Product>
-		<Version>4.1.0</Version>
+		<Version>4.1.1</Version>
```
This should be version `4.2.0` for a new feature, since I try to adhere to [semantic versioning](https://semver.org/) as best I can on a project that doesn't really have an API.

VariableVixen reviewed (changes requested):
A couple of changes but overall it looks fine. My only concern is that you said this is to help visualise your melee range; I really didn't pay much attention to in-game ranges when designing the distance handling for the guide rendering, so I don't know how much help it actually will be.

PrincessRTFM commented:
I also want to add that this is some solid work, I don't see anything that looks like it would break. You're right about it being a lot of arguments on the circle drawing function, but the only real alternative would be something like a `class`/`struct` to hold them, and that's unnecessary extra allocation that has to be done every frame. I think it works like this well enough. If you can confirm that the configurable distance is proportional to in-game distance (maybe add a tooltip on it for some common distances, or at least explaining the ratio?) then I think it's good to merge after those minor convention changes.

Nopileos2 commented:
From what I gather world coordinates are all in yalms. So From how I set it 10 units in the slider is 1 yalm in the game. I used the same division by 10 logic to have a int slider but a float value in the end. So 30 is 3 yalms. I set the default of the outer circle to 35 which is max melee range which seems to be calculated by the distance of the player circle to the target circle. The player target circle seems to be about 0.5 yalms in radius (not sure if 100% accurate but at least very close).

Testing on flat ground setting it to 35 very closely resembles when you can attack with a melee skill and when not. On non flat ground it can be quite weird to use because of angles, since the circle is only 2D and will thus clip through the ground or be in the air.

For me it helps since I now have the inner and outer circle as the region I can freely move and still be able to do melee skills. Being able to adjust the range helps if I do not want to orient myself on the absolute max distance.

VariableVixen committed:
Fix a memory leak

Just spotted that the two new pointers for the new sliders were allocated but never freed. Also shrunk the tooltip text while I was at it.

VariableVixen reviewed (approved):
Looks good now. Made a small fix for a memory leak I spotted, everything else should be okay.
